dowhy/causal_estimator.py

```diff
@@ -679,6 +685,91 @@ def signif_results_tostr(self, signif_results):
+    :param treatment_value: Value of the treatment in the treated group, for effect estimation. If treatment is multi-variate, this can be a list.
+    :param test_significance: Binary flag on whether to additionally do a statistical signficance test for the estimate.
+    :param evaluate_effect_strength: (Experimental) Binary flag on whether to estimate the relative strength of the treatment's effect. This measure can be used to compare different treatments for the same outcome (by running this method with different treatments sequentially).
+    :param confidence_intervals: (Experimental) Binary flag indicating whether confidence intervals should be computed.
+    :param target_units: (Experimental) The units for which the treatment effect should be estimated. This can be of three types. (1) a string for common specifications of target units (namely, "ate", "att" and "atc"), (2) a lambda function that can be used as an index for the data (pandas DataFrame), or (3) a new DataFrame that contains values of the effect_modifiers and effect will be estimated only for this new data.
+    :param effect_modifiers: Names of effect modifier variables can be (optionally) specified here too, since they do not affect identification. If None, the effect_modifiers from the CausalModel are used.
+    :param fit_estimator: Boolean flag on whether to fit the estimator.
+        Setting it to False is useful to estimate the effect on new data using a previously fitted estimator.
+    :param method_params: Dictionary containing any method-specific parameters. These are passed directly to the estimating method. See the docs for each estimation method for allowed method-specific params.
+    :returns: An instance of the CausalEstimate class, containing the causal effect estimate
+        and other method-dependent information
+
+    """
+    treatment = parse_state(treatment)
+    outcome = parse_state(outcome)
+    causal_estimator_class = method.__class__
+
+    identified_estimand.set_identifier_method(identifier_name)
+
+    if fit_estimator:
+        if identified_estimand.no_directed_path:
+            logger.warning("No directed path from {0} to {1}.".format(treatment, outcome))
+            return CausalEstimate(
+                0, identified_estimand, None, control_value=control_value, treatment_value=treatment_value
+            )
+        # Check if estimator's target estimand is identified
+        elif identified_estimand.estimands[identifier_name] is None:
+            logger.error("No valid identified estimand available.")
+            return CausalEstimate(None, None, None, control_value=control_value, treatment_value=treatment_value)
+    else:
+        # Estimator had been computed in a previous call
+        method.update_input(treatment_value, control_value, target_units)
+
+    estimate = method.estimate_effect()
```

Comment:
let's follow the sklearn protocol and provide "data" to the estimate_effect method. For backwards compatibility, the data parameter can be optional. Inside the estimate_effect method, we can check that if data parameter is None, then use self.data

Reply:
actually, to be fully consistent, let's provide X and y. Here y is the outcome variable, provided as `data[outcome_name]` and X would be data.drop(columns=["outcome_name"])

Reply:
Right now, this `estimate_effect` should also take fit_estimator param. But the ideal solution later is: 
```
 if fit_estimate: method.fit()
method.effect()
```

Reply:
Agree, I'll do this when I refactor all the estimator objects in a future PR, that's why I decided to leave it as it currently is